### ISSUE TITLE
Filter "Show only PRs" ignores times imported from Moposite

### ISSUE THREAD

**Issue by culinko** [elmadev/elmaonline-site#540]
Hi :) I have found something that looks like a bug so here we go.

If you go to your times and apply the filter "Show only PRs",  it will show the best times driven in EOL. However if by any chance you have an internal time imported from Moposite which is faster than any EOL time in that level, the Moposite time will be ignored and the worse EOL time will be considered as a "PR" and therefore shown in the list. Imo this is not correct and PR should always be only the best time in the level, no matter if driven in EOL or imported from Moposite. Therefore this (or any other) time for this level driven in EOL should not be shown when the filter is active. Adding images for better explanation:

Times driven in EOL with the filter applied:
<img width="839" alt="show-only-prs" src="https://user-images.githubusercontent.com/11700272/156431147-f3d14c27-6b4d-4c6d-b8c7-104a917675a5.png">

My best times (including times imported from Moposite):
<img width="435" alt="internal-times" src="https://user-images.githubusercontent.com/11700272/156431168-31a958d5-db08-4d0d-80e2-84eb58c90d34.png">

With the proposed change, the 2 times in first pic would not be displayed in this case. What do you think? Thanks :)

**Comment by j-masci** [comment]
the tab was also added with the in game feature to upload replays for every finish, so in that sense it makes a bit of sense to have it only show times driven in eol. But maybe it also makes sense to show legacy times properly. In general legacy times are often hard to deal with because of how the database is structured. For example, if you don't check "show only pr's" then it queries the allfinished table, if you do check it, it queries besttime (I assume), which holds every ones personal records on all levels. For legacy times, there is a legacybesttime table which holds only personal records for levelpacks (including internals) that are marked as having legacy times. But, there is no legacyallfinished table for example, as we just don't have that data. In other words, legacy times would then get shown only if you check "show only prs" which, perhaps is not really an issue because "legacy" finishes would have had to have happened like 10 years ago anyways.

Anyways I don't care either way. My only point is that legacy times in general, while they are a good feature for many things, are also not so trivial to include in every context.

**Comment by culinko** [comment]
Yeah, I totally agree with everything that you said.

I'm wondering, are there many such places where we would need to use the special treatment of including the legacy times in our queries? I personally don't see any other place which would be similarly important as this but I could very well be missing something as I'm not as familiar with all of the parts of the site yet.

**Comment by j-masci** [comment]
I think probably most relevant features do actually have legacy times support. I guess in this case maybe a separate option could be good for whether or not to include legacy times. Also the page currently lacks pagination which i'm sure it's just because sunehs put it together quickly as there was a bunch of other things that had to be done at the same timed for that game and website update. But I would assume actually that adding legacy times only when "show prs" is checked would be pretty easy? I haven't looked at the code but it seems likely you can just change the table you read from and that's about it. It's just that if you have a legacy time without a replay attached that's better than an EOL time driven after the game update, then it will be harder to find that replay. Maybe another option should be added: show only times with replays. But that also might get a bit complex if there's also a "show legacy times" toggle. Anyways I'm not against the change, just pointed out a possible explanation as to why it wasn't there in the first version of the page.

**Comment by culinko** [comment]
Yeah, I agree with having a separate toggle for including/not including the legacy times instead of changing the behavior of the  current "Show only PRs" toggle.

In order for the new toggle to be displayed, 2 conditions should be met:

1. The toggle "Show only PRs" is enabled (as you have already suggested)
2. The player actually has a legacy PR (as I don't think the new toggle serves any purpose if the player has no legacy records)

As for the replays, imho it's not really important if the PR has a replay attached to it, because a PR without a replay is still a PR. And currently EOL times without replays are also displayed.
